src/hugr/rewrite/simple_replace.rs

```diff
@@ -33,20 +31,31 @@ pub struct SimpleReplacement {
 impl SimpleReplacement {
     /// Create a new [`SimpleReplacement`] specification.
     pub fn new(
-        parent: Node,
-        removal: HashSet<Node>,
+        subgraph: SiblingSubgraph,
         replacement: Hugr,
         nu_inp: HashMap<(Node, Port), (Node, Port)>,
         nu_out: HashMap<(Node, Port), Port>,
     ) -> Self {
         Self {
-            parent,
-            removal,
+            subgraph,
             replacement,
             nu_inp,
             nu_out,
         }
     }
+
+    /// Number of nodes added to the hugr by this replacement.
+    ///
+    /// This will be negative if the replacement is a reduction in node count and
+    /// positive otherwise.
+    pub fn node_count_delta(&self) -> isize {
```

Comment:
You are right -- particularly since all fields were public. I'm removing this function. I'm also making the attributes private with the getters `fn subgraph` and `fn replacement`.